rts/System/FileSystem/ArchiveScanner.cpp

```diff
@@ -1315,20 +1320,44 @@ void CArchiveScanner::WriteCacheData(const std::string& filename)
 			allPoolRootDirs.emplace(CPoolArchive::GetPoolRootDirectory(ai.path + ai.origName));
 		}
 
-		for (auto it = poolFilesInfo.begin(); it != poolFilesInfo.end(); /*NOOP*/) {
-			// cleanup files that got deleted in the meantime
-			bool found = false;
+		const uint32_t seed = std::chrono::system_clock::now().time_since_epoch().count();
+		std::mt19937 generator(seed);
+		std::uniform_int_distribution<size_t> distribution(0, poolFilesInfo.size() - 1);
+		auto startOffset = distribution(generator);
+
+		auto st = poolFilesInfo.begin();
+		std::advance(st, startOffset);
+		auto it = st;
+
+		// we only want to check if the file still exists, we don't check for size / modDate
+		// this is checked in the checksum code anyway.
+		const auto ExistenceTest = [&allPoolRootDirs = std::as_const(allPoolRootDirs)](const auto& it) {
 			for (const auto& poolRootDir : allPoolRootDirs) {
-				if (FileSystem::FileExists(CPoolArchive::GetPoolFilePath(poolRootDir, it->first))) {
-					found = true;
-					break;
+				const auto fileName = CPoolArchive::GetPoolFilePath(poolRootDir, it->first);
+				if (FileSystem::FileExists(fileName)) {
```

Comment:
This is going to be slow doing syscall per file... I recommend stealing code from pr-downloader https://github.com/beyond-all-reason/pr-downloader/blob/185d0bb6a07e4cec56236d18f03d4cfe033e10f1/src/FileSystem/FileSystem.cpp#L299-L412

Intersecting files list in sdp with what is listed on disk takes milliseconds in prd.